src/providers/ogr/qgsogrprovider.cpp

```diff
@@ -1972,6 +1978,56 @@ bool QgsOgrProvider::doInitialActionsForEdition()
   return true;
 }
 
+#ifndef QT_NO_NETWORKPROXY
+void QgsOgrProvider::setupProxy()
+{
+  // Check proxy configuration, they are application level but
+  // instead of adding an API and complex signal/slot connections
+  // given the limited cost of checking them on every provider instantiation
+  // we can do it here so that new settings are applied whenever a new layer
+  // is created.
+  QgsSettings settings;
+  // Check that proxy is enabled
+  if ( settings.value( QStringLiteral( "proxy/proxyEnabled" ), false ).toBool() )
+  {
+    // Get the first configured proxy
+    QList<QNetworkProxy> proxyes( QgsNetworkAccessManager::instance()->proxyFactory()->queryProxy( ) );
+    if ( ! proxyes.isEmpty() )
+    {
+      QNetworkProxy proxy( proxyes.first() );
+      // TODO/FIXME: check excludes (the GDAL config options are global, we need a per-connection config option)
+      //QStringList excludes;
+      //excludes = settings.value( QStringLiteral( "proxy/proxyExcludedUrls" ), "" ).toString().split( '|', QString::SkipEmptyParts );
+
+      QString proxyHost( proxy.hostName() );
+      qint16 proxyPort( proxy.port() );
+
+      QString proxyUser( proxy.user() );
+      QString proxyPassword( proxy.password() );
+
+      if ( ! proxyHost.isEmpty() )
+      {
+        QString connection( proxyHost );
+        if ( proxyPort )
+        {
+          connection += ':' +  QString::number( proxyPort );
+        }
+        CPLSetConfigOption( "GDAL_HTTP_PROXY", connection.toUtf8() );
```

Comment:
If the proxy is set, it's mode is DefaultProxy, and we are on Windows, is it possible to set the GDAL GDAL_PROXY_AUTH  option to NTLM? @rouault does that make sense to you?